src/amber/dsl/router.cr

```diff
@@ -25,13 +25,16 @@ module Amber::DSL
       router.add(%route)
     end
 
+    macro namespace(scoped_namespace)
+      {{with DSL::Router.new(self, valve, scope + scoped_namespace) yield}}
```

Comment:
Clever.  💯

Reply:
Thanks !

But it not work :( 

The scope don't change.

Reply:
I tried 
```
      old_scope = scope
      new_scope = scope + {{scoped_namespace}}

      scope = new_scope
      {{yield}}
      scope = old_scope
```

But I got 

```
instantiating 'Amber::Router::Router#draw(Symbol, String)'
in spec/amber/dsl/server_spec.cr:138: macro didn't expand to a valid program, it expanded to:

================================================================================
--------------------------------------------------------------------------------
   1.       puts "scope before #{scope}"
   2.       old_scope = scope
   3.       new_scope = scope + "/test"
   4. 
   5.       scope = new_scope
   6.       begin get("/hello", HelloController, :index)
   7. namespace("/test2") do
   8.   get("/hello", HelloController, :index)
   9. end
  10.  end
  11.       scope = old_scope
  12. 
  13.       puts "scope after #{new_scope}"
  14.       # begin get("/hello", HelloController, :index)
  15. namespace("/test2") do
  16.   get("/hello", HelloController, :index)
  17. end
  18.  end
  19.     
--------------------------------------------------------------------------------
Syntax error in spec/amber/dsl/server_spec.cr:144: expecting token 'EOF', not 'end'

              end
 ^

================================================================================

              namespace "/test" do
              ^~~~~~~~~
```